userapi/storage/sqlite3/stats_table.go

```diff
@@ -0,0 +1,360 @@
+// Copyright 2022 The Matrix.org Foundation C.I.C.
```

Comment:
Needs tests. For SQLite you can pretty easily use `:memory:`. I'm not confident these queries work correctly without tests.